tools/ci_connector_ops/ci_connector_ops/pipelines/actions/secrets.py

```diff
@@ -57,7 +56,12 @@ async def download(context: ConnectorContext, gcp_gsm_env_variable_name: str = "
     # We don't want to print secrets in the logs when running locally.
     if context.is_ci:
         context.secrets_to_mask = await get_secrets_to_mask(with_downloaded_secrets)
-    return with_downloaded_secrets.directory(secrets_path)
+    connector_secrets = {}
+    for secret_file in await with_downloaded_secrets.directory(secrets_path).entries():
+        secret_plaintext = await with_downloaded_secrets.directory(secrets_path).file(secret_file).contents()
+        connector_secrets[secret_file] = context.dagger_client.set_secret(secret_file, secret_plaintext)
+
+    return connector_secrets
```

Comment:
This change allows cacheable secrets. Otherwise a new `directory` instance is returned when `download` happens, even if the downloaded secrets are the same and even if we fix their creation timestamp...
This logic is also better because connector secrets are actually considered as secrets by Dagger.